concrete/src/User/User.php

```diff
@@ -569,14 +569,17 @@ public function refreshUserGroups()
     }
 
     /**
-     * @return array
+     * Function to return Permission Access Entities belonging to this user object
+     *
+     * @return PermissionAccessEntity[] | array
      */
     public function getUserAccessEntityObjects()
     {
         $app = Application::getFacadeApplication();
         $session = $app['session'];
         $validator = $app->make(SessionValidator::class);
-        if ($validator->hasActiveSession()) {
+        // Check if the user is loggged in
+        if ($validator->hasActiveSession() && self::isLoggedIn()) {
```

Comment:
What about
```php
        if ($validator->hasActiveSession() && $this->isRegistered()) {
```
instead?

Reply:
Because here in this specific instance we don't want to check if this current user object has an ID greater than 0, we want to check if a User is logged in (not the current user object)

For example if your are logged in as User 1 and you have a page or a script that gets the user access entities for say User 2. What currently happens User 1 would have their entities overwritten.

If we changed it to isRegistered() all it is doing is saying this current user object has an ID greater than 0.. Though I can change this to something a bit better, because having a session doesnt mean you are logged